### ISSUE TITLE
Shopsanity LLR Deku Salescrubs

### ISSUE THREAD

**Issue by friedpen** [TestRunnerSRL/OoT-Randomizer#77]
The LLR Grotto appears to have two different instances, one for nighttime LLR and one for daytime LLR. Daytime contains the items specified within the spoiler log. The nighttime grotto appears to always have 40 Quiver, 40 Bomb Bag, and Silver Scale. Haven't been able to test SoT/SoS grottos but the Quiver+Bomb Bag+Silver Scale has popped up in another run previously where I had Silver Scale and it did not upgrade to Golden Scale. In this settings/seed the bomb bag is unusable without a base 30 and the bow from Bazaar can be upgraded to a 40.

Example settings/seed where this can be observed.
Settings : RJ6B8M5WWNN9MJA
Seed : thanksnintendo

**Comment by TestRunnerSRL** [comment]
Time of day, hmm. Thought I was handling that but I'll look into it. Thanks a lot for finding this. Will mention the getting those random ammo upgrade items are a filler item. You are never supposed to actually obtain them, you are getting them because they don't know to give you.

**Comment by friedpen** [comment]
There's another bug I've run into which is causing some Deku Salescrubs not to spawn. But so far I have not found a way to replicate the issue, even making a new file on the same settings/seed yields different results. Since the Deku Salescrubs appear in the world sometimes I have reason to believe its related to code so that you don't buy from the same scrub more than once. However, I believe the Deku Shield scrub is unique and I had an instance on this seed where it did not spawn. Making a second file, the scrub spawned as normal. I tried different times of day, child vs adult, doing LLR error grotto, and even checking LW Nut Salescrub before DC Nut Salescrub and I could not reproduce the bug on either Salescrubs.

**Comment by TestRunnerSRL** [comment]
Hmm, probably an issue with using the same flags as something else, most likely 2 scrubs sharing the same checks. Do you think you'd be able to find which sets are a problem?

**Comment by friedpen** [comment]
So far I've only seen DC Deku Shield, DC Nut, and JJ Nut issues. DCs occurred in a run with a long child 1 section where I had no access to bombs due to LLR glitched 30 bag. From that session I'm sure I only checked the 3 LW, and both sets of LLR (night/day). I believe I checked LLR night and LW after getting DC Nut in a different save and had no results. The run with JJ Nut I had not known of LLR day's existence and had no issues with all 4 DC. I believe I checked all bomb grottos but had no SoS/Epona/Longshot/Hammer in this prior run. I would guess JJ has some connection to Lake Hylia's gravestone grotto but I haven't had a chance to test JJ again.

**Comment by friedpen** [comment]
I cannot find a way to reproduce this through gameplay, however I recalled seeing a message when I first created the seed which mentioned overlapping records. No idea if this is related but it is worth bringing up possibly. Same settings/seed as above.

dmadata info
bfac30 c004e0 58b0
3485000 34881b8 31b8 <-- overlapping record
c02e00 c04410 1610
dmadata info
3471000 3471400 400
3489000 34b27b0 297b0 <-- overlapping record
3473000 3473010 10
dmadata info
347c000 347c010 10
3480000 3485000 5000 <-- overlapping record
347e000 347e040 40
